OsmAnd/src/net/osmand/plus/activities/SavingTrackHelper.java

```diff
@@ -212,11 +212,12 @@ public synchronized SaveGpxResult saveDataToGpx(File dir) {
 					log.debug("Filename: " + f);
 					File fout = new File(dir, f + IndexConstants.GPX_FILE_EXT);
 					if (!data.get(f).isEmpty()) {
+						Locale locale = Locale.getDefault();
 						WptPt pt = data.get(f).findPointToShow();
-						String fileName = f + "_" + new SimpleDateFormat("HH-mm_EEE", Locale.US).format(new Date(pt.time)); //$NON-NLS-1$
+						String fileName = f + "_" + new SimpleDateFormat("HH-mm_EEE", locale).format(new Date(pt.time)); //$NON-NLS-1$
 						Integer track_storage_directory = ctx.getSettings().TRACK_STORAGE_DIRECTORY.get();
-						if (track_storage_directory != OsmandSettings.REC_DIRECTORY) {
-							SimpleDateFormat dateDirFormat = new SimpleDateFormat("yyyy-MM");
+						if (!track_storage_directory.equals(OsmandSettings.REC_DIRECTORY)) {
```

Comment:
1. Wrong naming / code style - CAMEL Case should be used
2. ALWAYS: (good codestyle) check CONSTANT.equals(variable) i.e.  OsmandSettings.REC_DIRECTORY.equals(...)